content/en/docs/concepts/security/secrets-good-practices.md

```diff
@@ -0,0 +1,93 @@
+description: >
+  Principles and practices for good Secret management for cluster administrators and application developers.
+content_type: concept
+---
+
+<!-- overview -->
+
+{{<glossary_definition prepend="In Kubernetes, a Secret is an object that"
+term_id="secret" length="all">}}
+
+The following good practices are intended for both cluster administrators and
+application developers. Use these guidelines to improve the security of your
+sensitive information in Secret objects, as well as to more effectively manage
+your Secrets.
+
+<!-- body -->
+
+## Cluster administrators
+
+### Configure encryption at rest
+
+By default, Secret objects are stored unencrypted in {{<glossary_tooltip
+term_id="etcd" text="etcd">}}. You should configure encryption of your Secret
+data in `etcd`. For instructions, refer to [Encrypt Secret data at
+rest](/docs/tasks/administer-cluster/encrypt-data/).
+
+### Configure RBAC policies for Secrets
+
+When planning your {{<glossary_tooltip term_id="rbac" text="Role-based Access Control">}} [(RBAC)](/docs/reference/access-authn-authz/rbac/) policies,
+consider the following guidelines for `Secret` objects. You should also follow the other guidelines in [RBAC good practices](/docs/concepts/security/rbac-good-practices).
+
+*  Only allow cluster administrators to access `etcd` directly. This includes
+   read-only access.
+*  Restrict `watch` and `list` access to only privileged, system-level
+   components. 
+*  Use the `resourceNames` field in your RBAC rules to restrict access to
+   specific named `Secret` objects in namespaces, instead of providing access to
+   all Secret objects.
+*  Restrict permissions such as `get`, `patch`, `update`, `delete`, and `create`
+   for `Secret` objects based on the principle of least privilege for each application.
+
+Users who can create Pods that use a Secret, either directly or by using
+Kubernetes objects such as Deployments, can also read the contents of the
+Secret. The same user could then run a Pod that exposes the Secret data.
+
+TODO: (How to reduce this risk?)
+
+### Improve etcd management policies
+
+`etcd` is backed by persistent storage. You should wipe or shred this backing
+storage when it is no longer in use.
+
+If you have multiple `etcd` instances, configure SSL/TLS communication between
+the instances.
+
+### Configure access to external Secrets
+
+{{% thirdparty-content %}}
+
+You can use a third-party Secrets store provider to keep your Secrets outside
+your cluster. The [Kubernetes Secrets Store CSI
+Driver](https://secrets-store-csi-driver.sigs.k8s.io/) is a DaemonSet that lets
+the kubelet retrieve and mount Secrets from your external store into the Pods
+that need the data.
+
+For a list of supported providers, refer to [Providers for the Secret Store CSI
+Driver](https://secrets-store-csi-driver.sigs.k8s.io/concepts.html#provider-for-the-secrets-store-csi-driver).
+
+### Encrypt Secrets with an externally-managed key
+
+TBD
```

Comment:
Honestly, we should not provide guidance for this usecase IMO because this is out of scope of kubernetes project.